src/utils/workspace.ts

```diff
@@ -147,13 +147,18 @@ export async function ensureFolderIsOpen(fsPath: string, actionContext: IActionC
             vscode.workspace.updateWorkspaceFolders(openFolders.length, 0, { uri: uri });
         } else {
             await vscode.commands.executeCommand('vscode.openFolder', uri, openBehavior === OpenBehavior.OpenInNewWindow /* forceNewWindow */);
+            if (openBehavior === OpenBehavior.OpenInNewWindow) {
+                // extension host will not be restarted since a new instance of VS Code is opened, but we don't want deploy to continue
+                throw new UserCancelledError();
+            }
         }
 
         if (message) {
             // After we've opened the folder, throw the error message for the sake of telemetry
             actionContext.suppressErrorDisplay = true;
             throw new Error(message);
```

Comment:
I feel like your new logic is correct and this old logic that threw `new Error(message)` is incorrect. Right? Can we get rid of that?

Reply:
Well, that message would contain a message about requiring the user to open the folder to deploy which was sent to telemetry.  We'd be losing that for cases 1 and 2.